### PR TITLE
Git hooks: treating warnings as error

### PR DESCRIPTION
**Pull request by fredericmoulins**
The second step for the hooks. Should help catch imports or includes of missing ansible files.

Please find some more details in the commit messages.

### CODE REVIEW

File: git-hooks/pre-commit [hunk 1]
```diff
@@ -1,4 +1,7 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (arodier) [inline git-hooks/pre-commit:1]: Unless this is absolutely necessary, I prefer dash.
this is going to run on the CI server (albeit as Jenkins user), and I don't trust bash security records.

Comment (fredericmoulins) [reply git-hooks/pre-commit:1]: It's to be able to use pipefail (I had resolved to that in pre-push).

It started without it, maybe I can try again: it should just mean removing the log on stdout and report differently, which might be a good thing. I am going to try that and update the PR.

**Comment by arodier** [comment]
Thanks, testing it right now.

**Comment by fredericmoulins** [comment]
Ok, I force-pushed on the PR:
- `/bin/sh`, avoiding pipes ;
- reduced ansible-lint log on the console to errors and warnings only ;
- logs are gathered in one file for pre-commit, one for pre-push ;
- xargs is used in both cases to have ansible run on each file, so it can output errors or warnings for all files.

**Comment by arodier** [comment]
If you feel we reached the limit of gnu dash, maybe we can write some clean hook in ~~perl~~ python ;-). TBH, that's probably the best option on the long term.
CI is making good progress...

**Pull request merged**